src/item.rs

```diff
@@ -88,3 +95,146 @@ impl Item<'_> {
 		}
 	}
 }
+
+/// Type of discriminant used.
+#[cfg_attr(test, derive(Debug))]
+pub enum Discriminant {
+	/// The enum has only a single variant.
+	Single,
+	/// The enum uses the default representation and has only unit variants.
+	UnitDefault,
+	/// The enum uses the default representation but has a non-unit variant.
+	Default,
+	/// The enum uses a non-default representation and has only unit variants.
+	UnitRepr(Representation),
+	/// The enum uses a non-default representation and has a non-unit variant.
+	Repr(Representation),
+}
+
+impl Discriminant {
+	/// Parse the representation of an item.
+	pub fn parse(attrs: &[Attribute], variants: &Punctuated<Variant, Token![,]>) -> Result<Self> {
+		if variants.len() == 1 {
+			return Ok(Self::Single);
+		}
+
+		let mut has_repr = None;
+
+		for attr in attrs {
+			if attr.path().is_ident("repr") {
+				if let Meta::List(list) = &attr.meta {
+					let list =
+						list.parse_args_with(Punctuated::<Ident, Token![,]>::parse_terminated)?;
+
+					for ident in list {
+						if let Some(repr) = Representation::parse(&ident) {
+							has_repr = Some(repr);
+							break;
+						}
+					}
+				} else {
+					return Err(Error::repr(attr.span()));
+				}
+			}
+		}
+
+		let is_unit = variants.iter().all(|variant| match &variant.fields {
+			Fields::Named(fields) => fields.named.is_empty(),
+			Fields::Unnamed(fields) => fields.unnamed.is_empty(),
+			Fields::Unit => true,
+		});
```

Comment:
Why not use https://docs.rs/syn/latest/syn/enum.Fields.html#method.is_empty

Reply:
Done.